### PR TITLE
Dotenv

### PR DESCRIPTION
**Pull request by MimoMoreno**
## Summary
 * Adding dotenv.
 * Modifying config files to avoid repetition of code.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,3 +41,7 @@ node_modules
 
 # Code coverage
 ./coverage
+
+# dotenv
+.env
```
Comment (mishuagopian) [inline .gitignore]: Can't you replace these two with?
```js
.env*
```

File: config/index.js [hunk 2]
```diff
@@ -2,4 +2,29 @@ const ENVIRONMENT = process.env.NODE_ENV || 'development';
 
 const configFile = `./${ENVIRONMENT}`;
 
-module.exports = require(configFile).config;
+const config = {};
+config.common = {
```
Comment (mishuagopian) [inline config/index.js]: You can do these two in one line:
```js
const config = {
  common: {
    ...
  }
};
```

File: app.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ const express = require('express'),
   rollbar = require('rollbar'),
   morgan = require('morgan'),
   path = require('path'),
+  dotenv = require('dotenv'),
```
Comment (mpizzagalli) [inline app.js]: Aren't we missing the `dotenv.config()` call?

File: .eslintrc [hunk 4]
```diff
@@ -62,6 +62,7 @@
     "no-mixed-requires": 0,                                       // http://eslint.org/docs/rules/no-mixed-requires
     "no-multi-str": 0,                                            // http://eslint.org/docs/rules/no-multi-str
     "no-nested-ternary": 0,                                       // http://eslint.org/docs/rules/no-nested-ternary
+    "no-param-reassign": 0,                                       // https://eslint.org/docs/rules/no-param-reassign
```
Comment (gonziesc) [inline .eslintrc]: I'd like that `setConfig` return a new config object so we maintain immutability. Then in `config/index.js` we can do `const config = require(configFile).setConfig(baseConfig);` and export config. What do you think?

File: .gitignore (hunk 1 above)
Comment (MimoMoreno) [reply .gitignore]: By the way I wrote the `config` there's no need of having the `.env.test` file

**Comment by mishuagopian** [comment]
@MimoMoreno please, check our reviews. I agree with @mpizzagalli on the `.config()` call and READMEs description!

File: .eslintrc (hunk 4 above)
Comment (MimoMoreno) [reply .eslintrc]: I think that this rule (off) could be useful in several places. Maybe we should discuss it.

File: README.md [hunk 5]
```diff
@@ -33,6 +33,11 @@ Now, to start your app run ```npm start``` in the rootpath of the project. Then
 #### Environments
 By default, the environment will be **development**, but you can easily change it using the **NODE_ENV** environmental variable.
 
+#### Environment variables
+`Dotenv` is used for managing environment variables. They are stored in the `/config/.env` file. Take into account that the variables defined in the `bashrc` are not overrided.
+
```
Comment (mpizzagalli) [inline README.md]: I would add a simple example how it would look a .env file.

File: config/development.js [hunk 6]
```diff
@@ -1,26 +1,3 @@
...
-      header_name: 'authorization',
-      secret: process.env.NODE_API_SESSION_SECRET
-    },
-    rollbar: {
-      accessToken: process.env.ROLLBAR_ACCESS_TOKEN,
-      environment: process.env.ROLLBAR_ENV
-    }
-  }
+exports.setConfig = config => {
+  (config.environment = 'development'), (config.isDevelopment = true);
```
Comment (mishuagopian) [inline config/development.js]: I'd export this as the following:
```node
module.exports = {
  environment: 'development',
  isDevelopment: true
};
```

File: config/index.js [hunk 7]
```diff
@@ -1,5 +1,33 @@
...
+    },
+    rollbar: {
+      accessToken: process.env.ROLLBAR_ACCESS_TOKEN,
+      environment: process.env.ROLLBAR_ENV
+    }
+  }
+};
+
+require(configFile).setConfig(config);
+module.exports = config;
```
Comment (mishuagopian) [inline config/index.js]: And we can replace these two lines with:
```node
const customConfig = require(configFile);
module.exports = Object.assign(config, customConfig);
```

**Comment by gonziesc** [comment]
@MimoMoreno remember to squash the commits

File: config/index.js [hunk 8]
```diff
@@ -1,5 +1,54 @@
...
-module.exports = require(configFile).config;
+const isObject = variable => {
+  return variable instanceof Object;
+};
+
+/*
+ * Deep copy of source object into tarjet object.
+ * It does not overwrite properties.
+*/
+const assignObject = (target, source) => {
```
Comment (mpizzagalli) [inline config/index.js]: I actually don't remember what we agreed the past friday, are we going to leave this here or move it to `app/helpers/` folder? I would go with the second option.

Comment (gonziesc) [reply config/index.js]: Second option :+1:

Comment (mishuagopian) [reply config/index.js]: If I could choose, I'd leave it here 😇

**Pull request merged**